app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt

```diff
@@ -0,0 +1,198 @@
+            OutlinedTextField(
+                value = title,
+                onValueChange = { title = it },
+                label = { Text("Title") },
+                placeholder = { Text("Title") }
+            )
+
+            OutlinedTextField(
+                value = description,
+                onValueChange = { description = it },
+                label = { Text("Description") },
+                placeholder = { Text("Description") }
+            )
+
+            OutlinedTextField(
+                value = dateText,
+                onValueChange = {
+                    if(it.isDigitsOnly() && it.length <= 8){
+                        dateText = it
+                    }
+                },
+                label = { Text("Date") },
+                placeholder = { Text("01/01/1970") },
+                visualTransformation = DateVisualTransformation(),
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number)
+            )
+
+            OutlinedTextField(
+                value = location,
+                onValueChange = {  },
+                label = { Text("Location") },
+                placeholder = { Text("Location") }
+            )
+
+            Button(
+                enabled = location.isNotBlank() && title.isNotBlank() && description.isNotBlank()
+                        && dateText.isNotBlank(),
+                onClick = {
+                    try{
+
+                        val dateFormat =
+                            SimpleDateFormat("dd/MM/yyyy", Locale.getDefault()).apply {
+                                isLenient = false // strict date format
+                            }
+                        val date = dateFormat.parse(dateText)
+                        val calendar =
+                            GregorianCalendar().apply {
+                                time = date!!
+                                set(Calendar.HOUR_OF_DAY, 0)
+                                set(Calendar.MINUTE, 0)
+                                set(Calendar.SECOND, 0)
+                            }
+                        val finalDate = Timestamp(calendar.time)
+
+                        val activity = Activity(
+                            activityModelView.getNewUid(),
+                            title,
+                            description,
+                            placeholerLocation,
+                            finalDate,
+                            mapOf()
+                        )
+
+                        activityModelView.addActivity(activity)
+
+
+
+                    }catch (e: java.text.ParseException){
+                        Toast.makeText(
+                            context, "Invalid format, date must be DD/MM/YYYY.", Toast.LENGTH_SHORT)
+                            .show()
+                    }
+
+                }
+            ){
+                Text("Save")
+            }
+        }
+    }
+}
+
+class DateVisualTransformation: VisualTransformation {
```

Comment:
It is a great idea to add a class for automatically parsing the date with slashes (/) !

Reply:
Thank you very much